configure.ac

```diff
@@ -51,7 +51,11 @@ CURL_CHECK_OPTION_ARES
 CURL_CHECK_OPTION_RT
 
 XC_CHECK_PATH_SEPARATOR
-AX_CODE_COVERAGE
+dnl Check if gcc is being used before adding AX_CODE_COVERAGE
+AS_IF([ test "$GCC" = "yes" ], [AX_CODE_COVERAGE],
+  # not using GCC so pass a test below - CODE_COVERAGE_ENABLED_TRUE is not zero length
+  CODE_COVERAGE_ENABLED_TRUE='#'
+)
```

Comment:
Unfortunately this won't work as a generic solution to your problem.

configure.ac is used when I generate the release tarball and if I then happen to use gcc when I do that, it'll include that code in the generated configure file.

Besides, the problem with `AX_CODE_COVERAGE` is that is gnu-makeisms included, it doesn't actually have a problem with non-gcc...

Reply:
Actually, it is generic.
The configure script I generate using autoreconf does include the logic to create the code coverage tests - but only if CC=gcc when ./configure is called.

From my generated configure script:
...
+3484  { $as_echo "$as_me:${as_lineno-$LINENO}: result: $ac_cv_path_SED" >&5
 +3485  $as_echo "$ac_cv_path_SED" >&6; }
 +3486   SED="$ac_cv_path_SED"
 +3487    rm -f conftest.sed
 +3488
 +3489  if  test "$GCC" = "yes" ; then :
 +3490
.....
 +3624
 +3625
 +3626                                  if  test "$GCC" = "no" ; then :
 +3627
 +3628                          as_fn_error $? "not compiling with gcc, which is required for gcov code coverage" "$LINENO" 5
 +3629
......
 +3860  else
 +3861    # not using GCC so pass a test below - CODE_COVERAGE_ENABLED_TRUE is not zero length
 +3862    CODE_COVERAGE_ENABLED_TRUE='#'
 +3863
 +3864  fi
 +3865
...

* So, my changed logic starts at line 3489, the effect is that when GCC != yes - configure skips to line 3860.
* lines 3626 to 3629 are part of the m4 macro
* lines 3860-3864 are to be sure that a test later in configure succeeds. This seems to be the logic to say CODE_COVERAGE_ENABLED_TRUE is false. In any case, when both CODE_COVERAGE_ENABLED_TRUE and CODE_COVERAGE_ENABLED_FALSE are zero length, configure dies.

Reply:
footnote: maybe the old logic is better than what I propose here - i.e., something like what it was:
AS_IF([ test "$enable_code_coverage" = "yes" ], ...

as the macro verifies later if GCC != "no"

I just do not know if this has already been parsed. In any case, I am absolutely positive this is best fixed in configure.ac - one way, or the other.

Reply:
> configure.ac is used when I generate the release tarball and if I then happen to use gcc when I do that, it'll include that code in the generated configure file.
* afaik - a c-compiler is not used to generate "configure" from configure.ac (and other stuff) - that is done by the "autotools" (autoconf, automake, libtool, etc).
* my intent is that all the info from the AX_CODE_COVERAGE token is included in "./configure", but only being included in the Makefile generated (e.g., src/Makefile) when "./configure" when gcc is the active compiler and excluded when gcc is not the active compiler.
* maybe I am missing something - but I think "gcc", "xlc", "clang", etc. are only being used AFTER configure is already generated (for distribution).

Reply:
Per comments below - moved the AX_CODE_COVERAGE after the XC_CHECK_PROG_CC to be sure the status of GCC was either "yes" or "no".

Reply:
All tests now report success. I do not know how to trigger XXX so that your 'request' is 'replied' to.